### PR TITLE
chore: add configurable posix claims fields to master config [RM-398]

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-398]

## Description

Add the following fields to the master config, for saml and oidc:
- Agent UID 
- Agent GID
- Agent User Name
- Agent Group Name

For now, these fields are unused. Docs will be updated once the fields have an impact.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[RM-398]: https://hpe-aiatscale.atlassian.net/browse/RM-398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/config/oidc_config.go [hunk 1]
```diff
@@ -17,6 +17,10 @@ type OIDCConfig struct {
 	AutoProvisionUsers          bool   `json:"auto_provision_users"`
 	GroupsAttributeName         string `json:"groups_attribute_name"`
 	DisplayNameAttributeName    string `json:"display_name_attribute_name"`
+	AgentUID                    int    `json:"agent_uid"`
+	AgentGID                    int    `json:"agent_gid"`
+	AgentUserName               string `json:"agent_user_name"`
+	AgentGroupName              string `json:"agent_group_name"`
```
Comment (corban-beaird) [inline master/internal/config/oidc_config.go:23]: nit: Could we instead make these match `GroupAttributeName` & `DisplayNameAttributeName`, that way it's clear that these aren't the values for these fields, but instead the name of the attribute that can be passed in. Maybe something along the lines of `AgentUIDAttributeName`?

**Review by corban-beaird — APPROVED** [review]
One small naming nit, but besides that looks solid! Thank you so much for your work on this!

File: master/internal/config/oidc_config.go (hunk 1 above)
Comment (kkunapuli) [reply master/internal/config/oidc_config.go:23]: Absolutely! I was wondering about the naming; what is the meaning of `AttributeName`? Is it kinda like a "key" in a key/value pair?

Comment (corban-beaird) [reply master/internal/config/oidc_config.go:23]: Yep exactly!

**Pull request merged**